test/helper.go

```diff
@@ -0,0 +1,277 @@
+		placement.OCMPlacementLabel: TestPlacement,
+	}
+}
+
+func GetValidTestGatewaySpec() v1beta1.GatewaySpec {
+	return v1beta1.GatewaySpec{
+		GatewayClassName: TestDummyCRName,
+		Listeners: []v1beta1.Listener{
+			{
+				Name:     v1beta1.SectionName(ValidTestHostname),
+				Hostname: GetTestHostname(ValidTestHostname),
+				Protocol: v1beta1.HTTPSProtocolType,
+			},
+		},
+	}
+}
+
+func GetTestMZ() *v1alpha1.ManagedZoneList {
+	return &v1alpha1.ManagedZoneList{
+		Items: []v1alpha1.ManagedZone{
+			{
+				ObjectMeta: v1.ObjectMeta{
+					Namespace: TestNamespace,
+				},
+				Spec: v1alpha1.ManagedZoneSpec{
+					DomainName: TestDomain,
+				},
+			},
+		},
+	}
+}
+
+func GetTestDNSRecord() *v1alpha1.DNSRecordList {
+	return &v1alpha1.DNSRecordList{
+		Items: []v1alpha1.DNSRecord{
+			{
+				ObjectMeta: v1.ObjectMeta{
+					Name:      TestDomain,
+					Namespace: TestNamespace,
+				},
+			},
+		},
+	}
+}
+
+func GotExpectedError(expected string, got error) bool {
+	if expected == "" {
+		return true
+	}
+	return strings.Contains(got.Error(), expected)
+}
+
+func GetTime() *v1.Time {
+	now := v1.Now()
+	return &now
+}
+
+func AssertNoErrorReconciliation() func(res ctrl.Result, err error, t *testing.T) {
+	return func(res ctrl.Result, err error, t *testing.T) {
+		if err != nil || !res.IsZero() {
+			t.Errorf("failed. Expected no error and empty response but got: err: %s, res: %v", err, res)
+		}
+	}
+}
+
+func GetGCTestRequest() ctrl.Request {
+	return ctrl.Request{
+		NamespacedName: types.NamespacedName{
+			Name: TestDummyCRName,
+		},
+	}
+}
+
+func GetValidTestClient(initLists ...client.ObjectList) client.WithWatch {
+	return fake.NewClientBuilder().
+		WithScheme(GetValidTestScheme()).
+		WithLists(initLists...).
+		Build()
+}
+
+func GetValidTestScheme() *runtime.Scheme {
+	scheme := runtime.NewScheme()
+	_ = v1beta1.AddToScheme(scheme)
+	_ = corev1.AddToScheme(scheme)
+	_ = v1alpha1.AddToScheme(scheme)
+	return scheme
+}
+
+func GetBasicScheme() *runtime.Scheme {
+	scheme := runtime.NewScheme()
+	_ = v1beta1.AddToScheme(scheme)
+	_ = corev1.AddToScheme(scheme)
+	return scheme
+}
+
+func VerifyGatewayClassAcceptance(name string, want bool) func(c client.Client, res ctrl.Result, err error, t *testing.T) {
+	return func(c client.Client, res ctrl.Result, err error, t *testing.T) {
+		if err != nil {
+			t.Fatalf("unexpected error: %s", err)
+		}
+		class := &v1beta1.GatewayClass{}
+		err = c.Get(context.TODO(), client.ObjectKey{Name: name}, class)
+		if err != nil {
+			t.Fatalf("error getting gateway class from client: %s", err)
+		}
+		if want != gatewayClassIsAccepted(class) {
+			t.Fatalf("controller ignored or not accepted gateway class")
+		}
+	}
+}
+
+func gatewayClassIsAccepted(gatewayClass *v1beta1.GatewayClass) bool {
+	acceptedCondition := conditions.GetConditionByType(gatewayClass.Status.Conditions, string(v1beta1.GatewayConditionAccepted))
+	return (acceptedCondition != nil && acceptedCondition.Status == v1.ConditionTrue)
+}
```

Comment:
Same here, what is this in the test package for?

Reply:
it is gone now - reusing one from the gateway package